lib/p2p/Pool.ts

```diff
@@ -65,10 +65,18 @@ class Pool extends EventEmitter {
     }
 
     const peer = Peer.fromOutbound(socketAddress);
-    await this.connectPeer(peer);
+    await this.tryConnectPeer(peer);
     return peer;
   }
 
+  private tryConnectPeer = async (peer: Peer): Promise<void> => {
+    try {
+      return await this.connectPeer(peer);
+    } catch (err) {
+      this.logger.warn(`connectPeer failed: ${err}`);
```

Comment:
Maybe the error should be returned to make finding the reason why the connection failed easier.

Reply:
I don't think it should return the error, because this method is just about making `connectPeer` errors silent. We should just return what `connectPeer` returns. 
But you made me notice that it returns nothing, so i'll fix that.

`Peer` just throws an error on `ECONNREFUSED` or timeout. If the reason will be important for us, except for logging, we could use the thrown object.

Reply:
I thought that if you try to connect to a peer with `xucli` and it fails you are pretty clueless about the reason why it failed. Especially if you don't know that the error is in the logs.